control-plane-operator/controllers/hostedapicache/cache_controller.go

```diff
@@ -0,0 +1,90 @@
+package hostedapicache
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/kas"
+	"github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/manifests"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/util/workqueue"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+type HostedAPICacheReconciler struct {
+	client.Client
+
+	hostedAPICache *hostedAPICache
+}
+
+func RegisterHostedAPICacheReconciler(mgr ctrl.Manager, cacheCtx context.Context, cacheLog logr.Logger) (*HostedAPICacheReconciler, error) {
+	r := &HostedAPICacheReconciler{
+		Client:         mgr.GetClient(),
+		hostedAPICache: newHostedAPICache(cacheCtx, cacheLog, mgr.GetScheme(), mgr.GetRESTMapper()),
+	}
+
+	labelSelectorPredicate, err := predicate.LabelSelectorPredicate(metav1.LabelSelector{
+		MatchLabels: map[string]string{"hypershift.openshift.io/kubeconfig": "external"},
```

Comment:
why the external kubeconfig? we should be going through the service network

Reply:
For now it's hard-coded to use the external endpoint because I like to test with local processes against a remote cluster. Before merging it would need to use the service endpoint. Might want to make this configurable somehow for development (incl. auto-detection from the entrypoint based on whether the operator process is running in or out of cluster?)

Reply:
Resolved this by making it configurable through operator command arguments